### PR TITLE
CH-1164 Added drush command to get segments

### PR DESCRIPTION
**Pull request by acquiachrisnagy**
Also, added response code check to getSegments and more complete error handling/error reporting

### CODE REVIEW

File: acquia_lift_profiles/acquia_lift_profiles.drush.inc [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * Provides Acquia Lift Profiles drush commands.
+ */
+
+/**
+ * Implements hook_drush_command().
+ */
+function acquia_lift_profiles_drush_command() {
+  $items['acquia-lift-profiles-get-segments'] = array (
+    'description' => 'List segments',
+    'aliases' => array('alpgs'),
```
Comment (katbailey) [inline acquia_lift_profiles/acquia_lift_profiles.drush.inc]: Could we choose something a little more intuitive or easy to say - e.g. alp-seg?

File: acquia_lift_profiles/includes/acquia_lift_profiles.classes.inc [hunk 2]
```diff
@@ -294,7 +294,17 @@ class ALProfilesAPI {
...
+    if ($response->code != 200 || !isset($response->data)) {
+      if ($response->code != 200) {
+        $vars = array(
+          'rest_api_status' => $response->code,
+          'rest_api_status_line' => $response->status_message,);
+        $fail_msg_template = 'Could not get segments from Acquia Lift Profiles "{rest_api_status} {rest_api_status_line}"';
+        $this->logger()->log(PersonalizeLogLevel::ERROR, $fail_msg_template, $vars);
+        $fail_msg = str_replace(array('{rest_api_status}','{rest_api_status_line}'),
+          array($vars['rest_api_status'],$vars['rest_api_status_line']), $fail_msg_template);
+        throw new ALProfilesException($fail_msg);
```
Comment (katbailey) [inline acquia_lift_profiles/includes/acquia_lift_profiles.classes.inc:306]: We'll need all other calls to this method wrapped in a try/catch now that this throws an exception.

Comment (acquiachrisnagy) [inline acquia_lift_profiles/includes/acquia_lift_profiles.classes.inc:306]: All callers of the method are wrapped in a try/catch

Comment (katbailey) [inline acquia_lift_profiles/includes/acquia_lift_profiles.classes.inc:306]: Oh sorry - I didn't actually check, just assumed :-/
I am noticing some test failures though so maybe just the test needs to be fixed?
You can run the tests from the command line with a drush command:
`drush test-run ALProfilesAPITest --uri=http://your_lift_install.local --dirty`

Comment (katbailey) [inline acquia_lift_profiles/includes/acquia_lift_profiles.classes.inc:306]: Or you can run them from admin/config/development/testing and get nicer output ;) Just look under the Personalize group for Acquia Lift Profiles Unit Tests. (Make sure you have simpletest module enabled.)

**Comment by acquiachrisnagy** [comment]
I had to add the status_message to the mock http response object

File: acquia_lift_profiles/acquia_lift_profiles.drush.inc (hunk 1 above)
Comment (acquiachrisnagy) [inline acquia_lift_profiles/acquia_lift_profiles.drush.inc]: I renamed it to alp-seg

**Pull request closed without merging**